mkosi/__init__.py

```diff
@@ -7812,15 +7812,13 @@ def run_shell_setup(config: MkosiConfig, pipe: bool = False, commands: Optional[
         cmdline += [f"--bind={config.build_sources}:/root/src", "--chdir=/root/src"]
 
     if config.verb == Verb.boot:
```

Comment:
Shouldn't this be
```suggestion
    if config.verb == Verb.boot and (commands or config.cmdline):
```
since otherwise `commands or config.cmdline` might be `None`?

Reply:
`config.cmdline` is typed to never be `None`, so we'll just add no arguments in that case